zerver/webhooks/sonarqube/tests.py

```diff
@@ -0,0 +1,53 @@
+from zerver.lib.test_classes import WebhookTestCase
+
+
+class SonarqubeHookTests(WebhookTestCase):
+    STREAM_NAME = "SonarQube"
+    URL_TEMPLATE = "/api/v1/external/sonarqube?api_key={api_key}&stream={stream}"
+    FIXTURE_DIR_NAME = "sonarqube"
+
+    def test_analysis_success(self) -> None:
+        expected_topic = "test-sonar"
+
+        expected_message = """
+Project [test-sonar](http://localhost:9000/dashboard?id=test-sonar) analysis resulted in success.
+        """.strip()
+
+        self.check_webhook(
+            "success",
+            expected_topic,
+            expected_message,
+            content_type="application/x-www-form-urlencoded",
+        )
+
+    def test_analysis_error(self) -> None:
+        expected_topic = "test-sonar"
+
+        expected_message = """
+Project [test-sonar](http://localhost:9000/dashboard?id=test-sonar) analysis resulted in error:
```

Comment:
Maybe just "... analysis of branch {branch} resulted" as how to work the branch name in?

Reply:
Sure